### DISCUSSION TITLE
Filebeat integration wrong nginx log-format

### DISCUSSION THREAD

**Discussion by itsecbern** [ZoeyVid/NPMplus#1225] [Q&A] [answered]
We would like to integrate your NPMplus to our SIEM.
When enabling the access.log, and using the nginx, log phraser it gives the following error:

`Provided Grok expressions do not match field value: [[06/Nov/2024:22:22:06 +0100] rpm.MYDOMAIN.ch 141.195.95.210 0.005 \"GET / HTTP/1.1\" 200 4103 4652 - Mozilla/5.0 (compatible; PRTG Network Monitor (www.paessler.com); Windows)] `

This is due to a customazed log format. The Module only uses the standart nginx-access-log format.
A option to set the logformat via env. variable would be a good stuff here. :)

Alternatively as a nice to have, could be also to bundle a filebeat directly into your container-image.

**Comment by Zoey2936** [comment, marked as answer]
Hello, changing the logformat would break the current goaccess and crowdsec integration. Does filebeat not support using a different log format?
For the direct integration, can you maybe explain what it is and what it is used for/why it should be integrated?
But as short alternative/workaround for you, you can configure a second logfile using the files in /opt/npm/nginx_custom with the default log format

**Comment by Zoey2936** [comment]
https://discuss.elastic.co/t/help-parsing-custom-nginx-logs-using-filebeat-and-ingest-pipelines/349974/6

**Comment by itsecbern** [comment]
Good Morning Zoey, thank you for your research. The link you posted did I also found yesterday and i made a ned GROK pattern, that can parse your log-format of the NPMplus - but it only seems to work inside a GROK-Debugger, but not as ingest Pipline as described.

However if you want to integrate it, here is the working GROK:

`\[%{HTTPDATE:nginx.access.time}\] %{HOSTNAME:nginx.access.host} %{IP:nginx.access.remote_ip} %{NUMBER:nginx.access.response_time:float} "%{WORD:nginx.access.method} %{URIPATH:nginx.access.request} %{GREEDYDATA:nginx.access.protocol}" %{NUMBER:http.response.status_code} %{NUMBER:http.response.body.bytes} %{NUMBER:http.response.total.bytes} %{DATA:http.request.referrer} %{GREEDYDATA:user_agent.original}`

a other test log for you:

`[07/Nov/2024:09:38:50 +0100] access.MYDOMAIN.CH 141.195.95.210 0.000 "GET /app/element/templates/blank.html HTTP/2.0" 200 284 573 https://access.MYDOMAIN.CH/ Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36`

I don't know exactly, why it's not parsing it via elasticsearch pipeline. I will again test a few things, but if this also not works, i think i will give it a try with "/opt/npm/nginx_custom"

**Reply by Zoey2936** [reply]
did you got it working?

**Comment by Zoey2936** [comment]
I will convert it to a discussion, please report back if you find a way how to get it to work, so that other people also get it working if they want to use filebeat. The nginx_custom solution should work in any case.
